proxy/grpcproxy/cluster.go

```diff
@@ -15,20 +15,99 @@
 package grpcproxy
 
 import (
+	"fmt"
+	"sync"
+
 	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/naming"
 	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
 
 	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	gnaming "google.golang.org/grpc/naming"
 )
 
 type clusterProxy struct {
 	client *clientv3.Client
```

Comment:
@heyitsanthony How can we not use `clientv3.Client` when we need to use session via `clientv3.Client` in `register`? And I tried to use `clientv3.Cluster` but we have defined separate response type for `clientv3.Cluster.MemberList` so it would need additional type case to convert it back to `pb....`...